### ISSUE TITLE
Resource Harness delegate approval fails when delegate is already approved

### ISSUE THREAD

**Issue by arimaverick** [harness/terraform-provider-harness#104]
Hi,

### Terraform Version
1.0.0

### Affected Resource(s)
harness_delegate_approval

### Expected Behavior
When terraform reruns apply on already approved resource it should recognize that the delegate was already approved and do nothing.

### Actual Behavior
Terraform calls approval API however delegate is already approved and in enabled state hence approval resource is failed. 

### Steps to Reproduce
Approve a delegate with harness_delegate_approval resource and then rerun it. 

@abrahamkoshy

**Comment by micahlmartin** [comment]
I tested this and I'm not seeing the behavior described here.

```
terraform {
  required_providers {
    harness = {
      source = "harness/harness"
    }
  }
}

provider "harness" {
  # Configuration options
}

data "harness_delegate" "test" {
  name = "TestCreateDelegate_RyQi"
}

resource "harness_delegate_approval" "test" {
  delegate_id = data.harness_delegate.test.id
  approve     = true
}
```


Here's the results

```
Note: You didn't use the -out option to save this plan, so Terraform can't guarantee to take exactly these actions if you run "terraform
apply" now.
❯ terraform apply

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following
symbols:
  + create

Terraform will perform the following actions:

  # harness_delegate_approval.test will be created
  + resource "harness_delegate_approval" "test" {
      + approve     = true
      + delegate_id = "QPMrrIz4TUyGj44DcGAaew"
      + id          = (known after apply)
      + status      = (known after apply)
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

harness_delegate_approval.test: Creating...
harness_delegate_approval.test: Creation complete after 1s [id=QPMrrIz4TUyGj44DcGAaew]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
❯ terraform plan
harness_delegate_approval.test: Refreshing state... [id=QPMrrIz4TUyGj44DcGAaew]

No changes. Your infrastructure matches the configuration.

Terraform has compared your real infrastructure against your configuration and found no differences, so no changes are needed.
❯ terraform apply
harness_delegate_approval.test: Refreshing state... [id=QPMrrIz4TUyGj44DcGAaew]

No changes. Your infrastructure matches the configuration.

Terraform has compared your real infrastructure against your configuration and found no differences, so no changes are needed.

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

```

**Comment by arimaverick** [comment]
Hi,

Thanks for looking into this. I can confirm the delegate approval if is reapplied for the same delegate is working absolutely fine. However we also tested the resource for different delegate. So if the delegate is renamed which is needed to approve, then the resource errors out and reports that the delegate is already in Enabled state instead of approving the new harness delegate.
Error we are getting in that case is "delegate approval status has already been changed". 
Steps to reproduce the error:
- use delegate_approve resource to approve delegate-1 \
- `terraform apply`
- rename delegate-1 to delegate-2 (waiting for approval) 
- `terraform apply`

**Comment by micahlmartin** [comment]
ok I see. I think then I need to set the name field so that it forces it to create new resource. Let me take a look.

**Comment by arimaverick** [comment]
Hi @micahlmartin 
What we have observed after the update has failed, in the state file the delegate_id and id fields are different. delegate_id is updated with the new "delegate-1" and id is still referring to the old "delegate-1" and status still in "Enabled" state which is why I think Terraform is referring that the delegate-2 is already in Enabled state. 
We have also observed that if the delegate-2 if is approved outside Terraform's state's knowledge, then also the resource fails with the same error instead of skipping the resource execution. 
@abrahamkoshy
